### PR TITLE
[Term Entry] PHP Array Method: array_splice()

### PR DESCRIPTION
**Pull request by bccarruth927**
### Description

- Add the term entry for the array_splice method in PHP
- Closes issue #3049 

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [x ] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---
+Title: 'array_splice()'
+Description: 'Removes a set of elements from an array and replaces them with a new set of elements'
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:3]:
```suggestion
Description: 'Removes a set of elements from an array and replaces them with a new set of elements.'
```

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  - 'Web Development'
+Tags:
+  - 'Arrays'
+  - 'Functions'
+CatalogContent:
+  - 'learn/php'
+  - 'paths/computer-science'
+---
+
+The **`array_splice()`** method takes an array as it's input value and replaces elements within the array with new elements. The developer can specify within the method the index value
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:15]:
```suggestion
The **`array_splice()`** method takes an array as it's input value and replaces elements within the array with new elements. The developer can specify within the method the starting index value for the replacement values, and the length of the values to be replaced.
```

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+the replacement values begin as well as the length of index values to be replaced in the original array.
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:16]:
```suggestion
```

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+```pseudo
+array_splice($input_array, $starting_index, $replacement_length, $replacement_values)
+```
+
+The `array_splice()` method has two required parameters and two optional parameters.
+
+**Required Parameters**
+  - `$input_array`: Is the original array that will recieve the new elements.
+  - `$starting_index`: Is the index value that specifies where the replacement elements will start within the `$input_array`. If a positive number is provided the replacement will start at that
+  index value. 1 = second value of the `array`. If a negative number is provided the replacement will start at that number from the last index value. -3 = third value from last of the `array`.
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:29]:
```suggestion
  - `$input_array`: The original array that will recieve the new elements.
  - `$starting_index`: The index value that specifies where the replacement elements will start within the `$input_array`. 
  
 > **Note:** Positive and negative index values are valid arguments for the `$starting_index`. A negative number  counts from the last index value in the array (i.e. -3 equals third value from last).
```

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  **Optional Parameters**
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:31]: this header should not be indented

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    - `$replacement_length`: Is the length of values to be replaced within the `$input_array`. 2 = two values will be replaced within `$input_array` starting at `$starting_index`.
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:32]:
```suggestion
    - `$replacement_length`: The length of values to be replaced within the `$input_array`. 
```

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    - `$replacement_values`: Is an array that contains the new values that will be 'spliced' into the `$input_array`.
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:33]:
```suggestion
    - `$replacement_values`: An array that contains the new values that will be 'spliced' into the `$input_array`.
```

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      - **Note** If there is only one replacement value a `string` can be used in place of an `array`.
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:34]:
```suggestion
> **Note:** If there is only one replacement value a `string` can be used in place of an `array`.
```

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+  [2] => "dandelion yellow"
+  [3] => "forest green"
+  [4] => "blue"
+  [5] => "purple"
+)
+```
+
+## Codebyte Example
+
+This runnable codebyte example demonstrates a before (the array in it's original state) and after (the array after alteration by array_splice()) of an array.
```
Comment (caupolicandiaz) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:64]:
```suggestion
This runnable codebyte example demonstrates an alteration using `array_splice()`.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@bccarruth927 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by bccarruth927** [comment]
@caupolicandiaz I apologize for the delay in updating the PR. I have made a second commit with the recommended changes provided and am awaiting a secondary review.

**Review by caupolicandiaz — APPROVED** [review]
@bccarruth927 Looks good. This is ready for another review.

File: content/php/concepts/arrays/terms/array-splice/array-splice.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  - `$input_array`: The original array that will recieve the new elements.
+  - `$starting_index`: The index value that specifies where the replacement elements will start within the `$input_array`.
+
+  > **Note:** Positive and negative index values are valid arguments for the `$starting_index`. A negative number counts from the last index value in the array (i.e. -3 equals third value from last).
+
+**Optional Parameters**
+  - `$replacement_length`: The length of values to be replaced within the `$input_array`.
+  - `$replacement_values`: An array that contains the new values that will be 'spliced' into the `$input_array`.
+
+  > **Note:** If there is only one replacement value a `string` can be used in place of an `array`.
```
Comment (SSwiniarski) [inline content/php/concepts/arrays/terms/array-splice/array-splice.md:35]:
```suggestion
> **Note:** If there is only one replacement value a `string` can be used in place of an `array`.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@bccarruth927 I did a review and left a few comments.

**Comment by bccarruth927** [comment]
@SSwiniarski I have made the adjustments based on your suggestions. Ready for another review.

**Review by SSwiniarski — APPROVED** [review]
@bccarruth927 Looks good.

**Pull request merged**